src/utils/internal.ts

```diff
@@ -2378,3 +2380,58 @@ export function isV6Spec(specName: string, specVersion: number): boolean {
     (specName === 'polymesh_private_dev' && specVersion < 2000000)
   );
 }
+
+/**
+ * @hidden
+ */
+export async function prepareStorageForCustomType(
+  customType: string | BigNumber,
+  knownTypes: string[],
+  context: Context
+): Promise<Storage['customTypeData']> {
+  let customTypeData: Storage['customTypeData'];
+
+  if (customType instanceof BigNumber) {
+    const rawId = bigNumberToU32(customType, context);
+    const rawValue = await context.polymeshApi.query.asset.customTypes(rawId);
+
+    if (rawValue.isEmpty) {
+      throw new PolymeshError({
+        code: ErrorCode.DataUnavailable,
+        message:
+          'createNftCollection was given a custom type ID that does not have an corresponding value',
```

Comment:
Since you modified `assetType` in `CreateAssetParams` to support `BigNumber`, this message needs to be modified accordingly to depict if method invoked was `createNftCollection` or `createAsset`. 

Earlier the BigNumber on type was only support in NftCollection and not Asset creation